### DISCUSSION TITLE
QSPI configuration for ESP32S3 based projects, any examples?

### DISCUSSION THREAD

**Discussion by TKilmar** [zephyrproject-rtos/zephyr#77582] [Q&A]
Using QSPI for controlling an oled display has become quite normal for esp32 dev boards. However, it is not easy to understand how one need to configure the pinctrl and spi_cfg for QSPI. Are there any examples out there?  Any SMEs on the esp32 spi driver that could comment on this?

**Comment by LChes** [comment]
Hey, I am also looking into this as I have an ESP32s3 dev kit that is wired up with QSPI to the SPD2010 display controller. If anyone could provide some info on how to proceed with pinctrl assignments and any other considerations that would be great

**Comment by sadguitarius** [comment]
I'm also trying to figure out the same thing. Any progress? Looks like maybe the [MSPI API](https://docs.zephyrproject.org/latest/hardware/peripherals/mspi.html) could be adapted to make this work. Discussion [here](https://github.com/zephyrproject-rtos/zephyr/pull/71769) as well. Not very many examples for MSPI at the moment but it looks like it was designed to support this kind of application. I still can't understand the best approach here though. I guess write an MSPI driver for ESP32 based on the datasheet and a display driver that runs on top of that?
